app/src/androidTest/java/com/duckduckgo/app/browser/WebViewRequestInterceptorTest.kt

```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright (c) 2018 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser
+
+import android.net.Uri
+import android.support.test.InstrumentationRegistry
+import android.support.test.annotation.UiThreadTest
+import android.webkit.WebResourceRequest
+import android.webkit.WebResourceResponse
+import android.webkit.WebView
+import com.duckduckgo.app.analyticsSurrogates.AnalyticsSurrogates
+import com.duckduckgo.app.analyticsSurrogates.SurrogateResponse
+import com.duckduckgo.app.httpsupgrade.HttpsUpgrader
+import com.duckduckgo.app.trackerdetection.TrackerDetector
+import com.duckduckgo.app.trackerdetection.model.TrackingEvent
+import com.nhaarman.mockito_kotlin.*
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Test
+import org.mockito.ArgumentMatchers.anyString
+import org.mockito.Mock
+import org.mockito.MockitoAnnotations
+
+class WebViewRequestInterceptorTest {
+
+    private lateinit var testee: WebViewRequestInterceptor
+
+    @Mock
+    private lateinit var mockTrackerDetector: TrackerDetector
+    @Mock
+    private lateinit var mockHttpsUpgrader: HttpsUpgrader
+    @Mock
+    private lateinit var mockAnalyticsSurrogates: AnalyticsSurrogates
+    @Mock
+    private lateinit var mockRequest: WebResourceRequest
+
+    private lateinit var webView: WebView
+
+    @UiThreadTest
+    @Before
+    fun setup() {
+        MockitoAnnotations.initMocks(this)
```

Comment:
We seem to be mixing and matching approaches.  Nothing to do here, but we should all perhaps have a chat about what the standard approach should be (I prefer just creating them directly without annotations or using initMocks personally).